packages/slate-react/src/components/content.js

```diff
@@ -175,11 +175,16 @@ class Content extends React.Component {
     if (current) {
       if (
         (
+          !isBackward &&
+          focusNode == current.startContainer &&
```

Comment:
What was the need for these updates to these conditionals? The added `isBackward` check makes sense to me, and seems like a smart improvement. The other one I'm not sure I understand.

Reply:
Sorry, i messed up that part, i thought it's a bug, but it isn't, the native textarea works the same way:

![](https://user-images.githubusercontent.com/1759425/32553368-0275fd90-c497-11e7-9e7d-81f156ace267.gif)

select text backward by clicking on a text, and select with shift+click, and continue the selection with shift + right arrow

I will clean it up today